docs/untyped-plutus-core/representation/README.md

```diff
@@ -0,0 +1,73 @@
+# representation
+
+It is possible to unify a typed AST and an untyped one:
+
+```haskell
+data Typity
+    = Typed
+    | Untyped
+
+type family ToType (typity :: Typity) where
+    ToType 'Typed   = Type
+    ToType 'Untyped = ()
+
+-- See Note [Encoding].
+data Term (typity :: Typity)
+    = Constant Constant
+    | Builtin Builtin
+    | Var Name
+    | LamAbs Name (ToType typity) (Term typity)
+    | Apply (Term typity) (Term typity)
+    | typity ~ 'Typed => TyAbs TyName Kind (Term typity)
```

Comment:
Not really, just to match with the blogpost that I'm referencing there and because it seems to be less noisy than the GADTs version, which I had to use for the alternative version for technical reasons.